api4/post.go

```diff
@@ -521,6 +521,31 @@ func getPostThread(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	auditRec := c.MakeAuditRecord("getPostThread", audit.Fail)
+
+	rPost, err := c.App.GetSinglePost(c.Params.PostId, false)
+	if err != nil {
+		c.Err = err
+		return
+	}
+	auditRec.AddMeta("post", rPost)
+	hasPermission := false
+	if c.App.SessionHasPermissionToChannel(*c.AppContext.Session(), rPost.ChannelId, model.PermissionReadChannel) {
+		hasPermission = true
+	} else if !*c.App.Config().MessageExportSettings.EnableExport {
+		if channel, cErr := c.App.GetChannel(rPost.ChannelId); cErr == nil {
+			if channel.Type == model.ChannelTypeOpen &&
+				c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), channel.TeamId, model.PermissionReadPublicChannel) {
+				hasPermission = true
+			}
+		}
+	}
+
+	if !hasPermission {
+		c.SetPermissionError(model.PermissionReadChannel)
```

Comment:
What are your thoughts about this error? It could be a permissions error if the channel is private, but in the case of a public channel this isn't exactly a permissions error. It's a slightly different error related to business rules around needing to be a channel member if compliance is enabled. I'm wondering if it's worth us differentiating to disambiguate the two. Thoughts?

Reply:
@mkraft 

I think this is still a permissions error. By enabling the compliance export, the site admin is enforcing that a user has to belong to the channel to view a post/thread. Admins and users with elevated roles on the team or system can still view the post. So it's still in the realm of permissions, but one that is indirectly set by an admin.

Because it's a permissions issue, we need to return a 403, as opposed to other HTTP errors like Unauthorized (401), or Server Error (500).

Would an explicit permission like "PermissionReadPost" be more helpful? This permission wouldn't actually grant access to a post (or anything), it would just be a more helpful return value from the server.

Reply:
I guess you're right that it is an authorization error (403 Forbidden) in the sense that the user is not authorized to read the post (without joining the channel first). What's not sitting well with me is telling the user they're not authorized to read the channel due to lacking `permission_read_channel`. It's a public channel so they're not able to read the post because of a missing precondition (joining the channel) not because of a missing permission. I think if we can attribute this to a unique error that documents the reason (ex "due to compliance rules configured on this instance the channel must be joined before its posts can be read") then we should be good.

I don't think we should add a new permission.

Reply:
@mkraft done